OpenEdXMobile/src/main/java/org/edx/mobile/util/VideoUtil.java

```diff
@@ -65,25 +66,25 @@ public static boolean isVideoDownloadable(@NonNull VideoData video) {
      */
     @Nullable
     public static String getPreferredVideoUrlForDownloading(@NonNull VideoData video) {
+        String preferredVideoUrl = null;
         final VideoInfo preferredVideoInfo = video.encodedVideos.getPreferredVideoInfoForDownloading();
         if (preferredVideoInfo == null || video.onlyOnWeb) {
-            return null;
+            preferredVideoUrl = null;
+        } else if (!videoHasFormat(preferredVideoInfo.url, VIDEO_FORMAT_M3U8)) {
+            preferredVideoUrl = preferredVideoInfo.url;
         }
-        if (!videoHasFormat(preferredVideoInfo.url, VIDEO_FORMAT_M3U8)) {
-            return preferredVideoInfo.url;
-        }
-        if (!new Config(MainApplication.instance()).isUsingVideoPipeline()) {
+        if (TextUtils.isEmpty(preferredVideoUrl) && !new Config(MainApplication.instance()).isUsingVideoPipeline()) {
             /**
```

Comment:
Following comment should be updated now to this for clear approach.
/*
              If {@link Config#USING_VIDEO_PIPELINE} feature flag is disabled, try to find some .mp4
              format url from {@link VideoData#allSources} urls and consider it for download.
             */